corpus/accounts/views.py

```diff
@@ -37,6 +41,34 @@ def signin(request):
                 return redirect("index")
             else:
                 return redirect(next)
+        else:
+            username = form.cleaned_data.get("username")
+            password = form.cleaned_data.get("password")
+
+            if username.endswith("@nitk.edu.in"):
+                exec_member = ExecutiveMember.objects.filter(edu_email=username)
+            elif username.endswith("@ieee.org"):
+                exec_member = ExecutiveMember.objects.filter(ieee_email=username)
+            elif re.search(r"^[1-9][0-9]{2}[A-Za-z]{2}[0-9]{3}$", username) is not None:
+                exec_member = ExecutiveMember.objects.filter(
+                    roll_number=username.upper()
+                )
+            elif re.search(r"^[0-9]{6,}$", username) is not None:
+                exec_member = ExecutiveMember.objects.filter(reg_number=username)
+            else:
+                exec_member = []
+
+            if len(exec_member) == 1:
```

Comment:
Instead of this, let's use `exec_member is not None`. Looks more intuitive (and pythonic xD) .